app/models/work_packages/scopes/covering_dates_and_days_of_week.rb

```diff
@@ -68,29 +70,29 @@ def query(days_of_week)
             GREATEST(work_package_start_date, work_package_due_date) AS end_date
           FROM work_packages_with_dates
         ),
-        -- expand period into days of the week. Limit to 7 days (more would be useless).
-        work_packages_days_of_week AS (
-          SELECT id,
-            extract(
-              isodow
-              from generate_series(
-                  work_packages_periods.start_date,
-                  LEAST(
-                    work_packages_periods.start_date + 6,
-                    work_packages_periods.end_date
-                  ),
-                  '1 day'
-                )
+        -- expand period into days.
+        work_packages_expaned_periods AS (
+          SELECT id, generate_series(
+            work_packages_periods.start_date,
+            work_packages_periods.end_date,
+            '1 day') AS dates,
+            -- limit to 7 days (more would be useless).
+            extract(isodow from generate_series(
+              work_packages_periods.start_date,
+              LEAST(
+                work_packages_periods.start_date + 6,
+                work_packages_periods.end_date
+              ),
+              '1 day')
```

Comment:
This `generate_series` for the day of the week is no longer necessary in this way. Before, its purpose was to turn the start and end dates of a work package entry into a number of entries (7 at most) with the days of the week they cover.

But since this change, a similar thing is done for all the dates between the the start and due date. Since that result is available as well, the next step can be to calculate the `isodow` for each date. 

That calculation is more efficient then calculating both series multiple times. On a larger database, this query as it is runs for about 7s which can be reduced to about 2 by using this:


```sql
...
-- All days between the start date of a work package and its due date
covered_dates AS (
  SELECT 
   id,
   generate_series(work_packages_periods.start_date,
                   work_packages_periods.end_date,
                   '1 day')          AS date
  FROM work_packages_periods
),
-- All days between the start date of a work package and its due date including the day of the week for each date													 
covered_dates_and_wday AS (
  SELECT 
    id, 
    date, 
    EXTRACT(isodow FROM date) dow
	FROM covered_dates
) 
-- select id of work packages covering the given days
SELECT id FROM covered_dates_and_wday
        WHERE dow IN (:days_of_week) OR dates IN (:dates)
```

Aside from the speedup, it also simplifies the code IMO. Since the `covered_dates` is now quite simple, the intermediary CTE `work_packages_periods ` might actually be removed by now in favor of coalescing within `covered_dates`.

Reply:
The `work_packages_with_dates` CTE contains an `ORDER BY work_packages.id` which is unnecessary and in fact harmful. I am aware that it was not introduced in this PR but removing it will increase the performance significantly.

Reply:
Done f350323156e39c12eff5a82c4efcd799ed9a4f71